servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/SniTest.java

```diff
@@ -97,9 +100,30 @@ void noSniClientDefaultServerFallbackSuccess() throws Exception {
         sniDefaultFallbackSuccess(serverContext -> HttpClients.forSingleAddress(serverHostAndPort(serverContext))
                 .sslConfig(new ClientSslConfigBuilder(DefaultTestCerts::loadServerCAPem)
                         .peerHost(serverPemHostname()).build())
+                .inferSniHostname(false)
                 .buildBlocking());
     }
 
+    @Test
+    void noSniClientDefaultServerFallbackFailExpected() throws Exception {
+        try (ServerContext serverContext = HttpServers.forAddress(localAddress(0))
+                .sslConfig(
+                        untrustedServerConfig(),
+                        singletonMap(InetAddress.getLoopbackAddress().getHostName(), trustedServerConfig())
+                )
+                .listenBlockingAndAwait((ctx, request, responseFactory) -> responseFactory.ok());
+             BlockingHttpClient client = HttpClients.forSingleAddress(
+                     HostAndPort.of(
```

Comment:
nit: consider removing `HostAndPort.of(` and use the method overload that takes `String, int`